packages/checkout/text-input/text-input.tsx

```diff
@@ -49,18 +53,18 @@ const TextInput = forwardRef< HTMLInputElement, TextInputProps >(
 				/* Do nothing */
 			},
 			feedback,
+			inputProps,
 		},
 		ref
 	) => {
 		const [ isActive, setIsActive ] = useState( false );
-
 		return (
 			<div
 				className={ classnames(
 					'wc-block-components-text-input',
 					className,
 					{
-						'is-active': isActive || value,
+						'is-active': isActive || value || value === 0,
```

Comment:
If this `value === 0` only reffers to the number type than I'd add that in the condition as well: `value === 0 && type === 'number'`